hs_core/models.py

```diff
@@ -3632,6 +3795,47 @@ def serializer(self):
         from .views.resource_metadata_rest_api import CoreMetaDataSerializer
         return CoreMetaDataSerializer(self)
 
+    def rdf_subject(self):
+        from .hydroshare import current_site_url
+        return URIRef("{}/resource/{}".format(current_site_url(), self.resource.short_id))
+
+    def rdf_metadata_subject(self):
+        from .hydroshare import current_site_url
+        return URIRef("{}/resource/{}/data/resourcemetadata.xml".format(current_site_url(), self.resource.short_id))
+
+    def rdf_type(self):
+        return HSTERMS.resource
+
+    def ingest_metadata(self, graph):
+        super(CoreMetaData, self).ingest_metadata(graph)
+        subject = self.rdf_subject_from_graph(graph)
+        extra_metadata = {}
+        for o in graph.objects(subject=subject, predicate=HSTERMS.extendedMetadata):
+            key = graph.value(subject=o, predicate=HSTERMS.key).value
+            value = graph.value(subject=o, predicate=HSTERMS.value).value
+            extra_metadata[key] = value
+        res = self.resource
+        res.extra_metadata = copy.deepcopy(extra_metadata)
+        res.save()
+
+    def get_rdf_graph(self):
+        graph = super(CoreMetaData, self).get_rdf_graph()
+
+        subject = self.rdf_subject()
+
+        # add any key/value metadata items
+        if len(self.resource.extra_metadata) > 0:
+            for key, value in list(self.resource.extra_metadata.items()):
```

Comment:
Not sure why ```self.resource.extra_metadata.items()``` needs to be converted into list before iterating with key and value. Is this really intended?

Reply:
it doesn't need to be converted to a list, removed.